nanomq/apps/broker.c

```diff
@@ -945,18 +945,17 @@ broker(conf *nanomq_conf)
 			log_error("nng_exchange_client_open failed %d", rv);
 		} else {
 			void *ex;
-			for (int i = 0; i < cvector_size(node->ex_list); i++) {
-				ex = node->ex_list[i];
-				nng_socket_set_ptr(
-				    *node->sock, NNG_OPT_EXCHANGE_ADD, ex);
-			}
+			ex = node->exchange;
+			nng_socket_set_ptr(*node->sock, NNG_OPT_EXCHANGE_BIND, ex);
 		}
 		log_debug("exchange %d init finished!\n", i);
 	}
 	// Webhook service
 	if (nanomq_conf->web_hook.enable) {
 		log_debug("Webhook service initialization");
 		start_webhook_service(nanomq_conf);
+		if (nanomq_conf->exchange.count > 0)
+			hook_exchange_init(nanomq_conf);
```

Comment:
??? should not be here
num_ctx is changed in the following init section

Reply:
and exchange is not related to webhook

Reply:
Now exchange is an independent service to start